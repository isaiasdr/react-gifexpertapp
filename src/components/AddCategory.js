import React, { useState } from 'react';
import propTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories( categories => { 
                if(!categories.find( categorie => categorie === inputValue))    
                    return [ inputValue, ...categories ];

                else
                    return [...categories];
            });

            setInputValue( '' );
        }
    }

    return (
        <form onSubmit = { handleSubmit }>
            <input 
                type="text"
                value = { inputValue }
                onChange = { handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: propTypes.func.isRequired
};