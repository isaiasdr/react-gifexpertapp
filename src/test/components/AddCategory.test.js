import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Tests in <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories} />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories} />);
    });

    test('should print correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should change text box', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', { target: {value} });
    });

    test('should not post information on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('should should call setCategories and clear the text box', () => {
       const input = wrapper.find('input');
       const value = "test submit";
       input.simulate('change', { target: {value} });
       wrapper.find('form').simulate('submit', { preventDefault(){} });
       expect( setCategories ).toHaveBeenCalled();
       expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
       expect( wrapper.find('input').prop("value") ).toBe('');
    });
});
