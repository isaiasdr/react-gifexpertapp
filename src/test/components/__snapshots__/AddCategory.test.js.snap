// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory /> should print correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
